product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/domain/SessionRepository.kt

```diff
@@ -0,0 +1,23 @@
+package com.walletconnect.web3.modal.domain
+
+import android.content.SharedPreferences
+
+private const val SESSION_TOPIC = "session_topic_key"
+private const val SELECTED_CHAIN = "selected_chain_key"
+
+internal class SessionRepository(
+    private val sharedPreferences: SharedPreferences
+) {
+
+    fun saveSessionTopic(topic: String) = sharedPreferences.edit().putString(SESSION_TOPIC, topic).apply()
```

Comment:
just FYI, you can also use the `.edit` function so you don't have to apply each time

```suggestion
    fun saveSessionTopic(topic: String) = sharedPreferences.edit { putString(SESSION_TOPIC, topic) }
```